src/store/reducer.js

```diff
@@ -5,13 +5,22 @@ import booksReducer from './books/books.reducer';
 import booksSearchReducer from './search/search.reducer';
 import loginReducer from './login/login.reducer';
 import userReducer from './user/user.reducer';
+import { LOGOUT_ACTIONS } from './logout/logout.actions';
 
-const rootReducer = combineReducers({
+const appReducer = combineReducers({
     booksCollection: booksReducer,
     booksSearch: booksSearchReducer,
     account: accountReducer,
     login: loginReducer,
     user: userReducer,
 });
 
+const { LOG_OUT } = LOGOUT_ACTIONS;
+const rootReducer = (state, action) => {
+    if (action.type === LOG_OUT) {
+        state = undefined;
+    }
+    return appReducer(state, action);
+};
+
 export default rootReducer;
```

Comment:
I really don't like this

Reply:
1. Why is the rootReducer renamed to appReducer
2. What are actions doing the main root reducer file
3. Why is there a reducer defined within this file and not in a dedicated one

Reply:
i changed it back to the old setup and created logoutsaga that dispatch multiple reset actions for the login and user data